app/controllers/register/company/AddCompanyDirectorsController.scala

```diff
@@ -78,4 +84,26 @@ class AddCompanyDirectorsController @Inject() (
         )
       }
   }
+
+  private def directorsWithFlag(
+                                 withFlag: Seq[(DirectorDetails, Boolean)],
+                                 directors: Seq[DirectorDetails],
+                                 flags: Seq[Boolean]
+                               ): Seq[(DirectorDetails, Boolean)] = {
+
+    if(directors.isEmpty){
+      withFlag
+    } else if (flags.isEmpty) {
+      val addWithFlag: (DirectorDetails, Boolean) = (directors.head, false)
```

Comment:
We don't really need to load this recurively, we could just assess  whether if we have `flags` beforehand and then just map over directors to return Seq[(Directors,boolean)]